remove includes from header / create required includes file / set title variable before include header

ivailo-devrix opened the pull request:
Refactoring header inclusion

Moved session opening functionality - from your comment in PR8
Create required includes file "required-includes.php"
Remove includes from header and move in includes/required-includes.php

Set "meta title" variable before include header

ivailo-devrix committed:
remove-includes-from-header/create-required-includes-file/submit-title-variable-before-include-header

ivailo-devrix committed:
Merge branch 'main' of https://github.com/yanachkov-devix/JobriX into feature/add-meta-title/refactoring-includes

Neychok reviewed (changes requested):
Can you remove the `test.php` file and fix the merge conflict. We should be good to merge it after that is fixed.

ivailo-devrix committed:
remove test

metodiew commented on job-view.php:
```diff
@@ -16,20 +8,31 @@
+                        <header class="job-header">
+                            <h2 class="job-title"><?php echo $row['title'] ?></h2>
+                            <div class="job-meta">
+                                <a class="meta-company" href="#"><?php echo $row['company_name'] ?></a>
```
the link here is not going to work?

ivailo-devrix replied:
There is no suitable HTML page to link to. There is no suitable HTML template. I'm not sure exactly how to solve the problem with this empty link, so I've left it empty.

metodiew replied:
just check if `$row['company_name'] ` is empty - if yes, do no display the `<a>` element

ivailo-devrix replied:
Fixed: For the given companies without a website, the name will not be displayed as a link.

```
if ( ! empty( $job['company_site'] ) ) { ?>
    <a class="meta-company" href="<?php echo 'http://' . $job['company_site'] ?>"><?php echo $job['company_name'] ?></a>
<?php } else {
	echo $job['company_name'];
}

```

metodiew commented on job-view.php:
```diff
@@ -16,20 +8,31 @@
+                <div class="job-card">
+                    <div class="job-primary">
+                        <header class="job-header">
+                            <h2 class="job-title"><?php echo $row['title'] ?></h2>
```
we are displaying the `$row` items relying that we are going to have these fields for each job? the comments goes for al `$row` variable

ivailo-devrix replied:
I created an empty array for all the elements that will be displayed.
Then for each I set the value from the DB if one exists.
This prevents an error from being thrown if an element is missing


$job = array(
	'title'               => '',
	'company_name'        => '',
	'company_description' => '',
	'company_site'        => '',
	'publication_date'    => '',
	'jobs_location'       => '',
	'salary'              => '',
	'description'         => '',
	'id_user'             => '',
);

foreach ( $row as $key => $value ) {
	if ( ! empty( $value ) && array_key_exists( $key, $job ) ) {
		$job[ $key ] = $value;
	}
}

metodiew reviewed (changes requested):
left a few inline comments, et's fix them

ivailo-devrix committed:
Merge branch 'main' into feature/add-meta-title/refactoring-includes

ivailo-devrix committed:
fix for problem: if there is no data in the database for a given value

$job = array(
	'title'               => '',
	'company_name'        => '',
	'company_description' => '',
	'company_site'        => '',
	'publication_date'    => '',
	'jobs_location'       => '',
	'salary'              => '',
	'description'         => '',
	'id_user'             => '',
);

foreach ( $row as $key => $value ) {
	if ( ! empty( $value ) && array_key_exists( $key, $job ) ) {
		$job[ $key ] = $value;
	}
}

yanachkov-devix commented:
@metodiew I have fixed everything from the comments.
New changes have been uploaded.

metodiew commented on job-view.php:
```diff
@@ -1,91 +1,122 @@
+								<?php
+								if ( ! empty( $job['company_site'] ) ) { ?>
+                                    <a class="meta-company"
+                                       href="<?php echo 'http://' . $job['company_site'] ?>"><?php echo $job['company_name'] ?></a>
```
the formatting of this is a bit weird, but what is more questionable is - why do we have `http://` and then we are getting `$job['company_site']`? What would happen if `$job['company_site']` is actually `https://devrix.com`?

ivailo-devrix replied:
$job['company_site'] always returns a domain. It never returns a url. In the registration form, we ask the client to leave a site name: doman.com or www.doman.com

metodiew commented on job-view.php:
```diff
@@ -1,91 +1,122 @@
+	                                        echo "Posted " . date_difference( $job['publication_date'] ) . " days ago";
                                         } else {
-                                            echo "Posted over 1 month ago";
+	                                        echo "Posted over 1 month ago";
```
why we didn't take care of this in the function and have just: `date_difference( $job['publication_date'] ` and in the function we can take care of how long the job was posted?

ivailo-devrix replied:
fixed

metodiew commented on job-view.php:
```diff
@@ -1,91 +1,122 @@
                             </div>
+                        </header>
+                        <div class="job-body">
+							<?php
```
formatting alert

ivailo-devrix replied:
I reformatted it. Not sure if this should be the end result but i think it's better.

metodiew commented on job-view.php:
```diff
@@ -1,91 +1,122 @@
+            <aside class="job-secondary">
+                <div class="job-logo">
+                    <div class="job-logo-box">
+						<?php
```
formatting

ivailo-devrix replied:
I reformatted it. Not sure if this should be the end result but i think it's better.

metodiew commented on job-view.php:
```diff
@@ -1,91 +1,122 @@
+                <div class="job-logo">
+                    <div class="job-logo-box">
+						<?php
+						$img_file_logo = "./img/company/" . $job['id_user'] . ".jpg";
```
what would happen if the image is not `.jpg`, but a `.png` file?

ivailo-devrix replied:
We can't have PNG or any other graphic format.
The idea is to keep some consistency.
When a client uploads another format other than JPG it will be converted on upload

metodiew commented on job-view.php:
```diff
@@ -1,91 +1,122 @@
+						?>
+                    </div>
+                </div>
+                <a href="#" class="button button-wide">Apply now</a>
```
formatting

ivailo-devrix replied:
I reformatted it. Not sure if this should be the end result but i think it's better.

metodiew reviewed (changes requested):
I did one more check of the changed files and left a number of inline comments, we have more work to do here

ivailo-devrix committed:
move if statement in date_difference( ) function

ivailo-devrix committed:
formatting fix

yanachkov-devix commented:
I fixed the code based on some of the comments.
I have only written a response to other comments about why I decided to choose this logic.

metodiew reviewed (approved):
synced on Slack - we'll be merging these and @yanachkov-devix will create a few tasks to update and do some refactoring

ivailo-devrix committed:
Merge branch 'main' of https://github.com/yanachkov-devix/JobriX into feature/add-meta-title/refactoring-includes
